### PR TITLE
STCON-150 Support tenant override in okapi resources

### PR DESCRIPTION
**Pull request by NikitaSedyx**
purpose: in scope of consortia work we need to fetch resources from different tenants, but there are some legacy class components that use manifests and stripes-connect. There is an option to override to hooks approach, but there are cases when it's required to override whole plugin (ui-plugin-find-user)

approach: introduce new optional okapi resource property to define tenant to be passed in headers. Usage can be reviewed in related PR

related to https://github.com/folio-org/ui-plugin-find-user/pull/238

### CODE REVIEW

**Review by alisher-epam — APPROVED** [review]

**Review by usavkov-epam — APPROVED** [review]

**Review by zburke — APPROVED** [review]
The change itself looks fine, and I really appreciate the updated API documentation!

Is there a way to write a test for this, e.g. configure a dummy-component with a `stripes` object with tenant `foo` that gets overridden in the manifest with tenant `bar`?

**Comment by NikitaSedyx** [comment]
@zburke I'll create a separate ticket for test as stripes-connect doesn't have any tests and even config

**Comment by zburke** [comment]
@NikitaSedyx, [stripes-connect has many tests](https://github.com/folio-org/stripes-connect/blob/master/test/integration.js) with coverage > 80% on most components. They just aren't written in Jest. 

I did approve this so you can merge this if you really want to, but it is always preferable that tests be included in the PRs that include the code changes, rather than being considered as a distinct task.

**Comment by NikitaSedyx** [comment]
@zburke thank you pointing this, to be honest I missed tests because I didn't find `jest`, but as soon as `stripes-connect` has tests I'll update PR with them before merging

**Pull request merged**
